src/main/java/org/opentripplanner/routing/RoutingService.java

```diff
@@ -53,7 +53,7 @@ public RoutingService(Graph graph) {
     // TODO We should probably not have the Router as a parameter here
     public RoutingResponse route(RoutingRequest request, Router router) {
         try {
-            var zoneId = graph.getTransitLayer().getTransitDataZoneId();
+            var zoneId = graph.getTimeZone().toZoneId();
```

Comment:
Are we sure this do not create a new problem if these timezone differ? There should only be one truth/source.

Reply:
What do you suggest? Only change the code in case where getTransitLayer returns null? And copy the logic of handling null over for the other case?
```suggestion
            var zoneId = graph.getTransitLayer() ? graph.getTransitLayer().getTransitDataZoneId() : TimeZone.getTimeZone("GMT");
```

Reply:
I think I will just approve your initial suggestion. I looked at the code now, and I see that we copy the TimeZone from the graph into the TransitLayer, so I think it is ok. This is probably something that I would like to take a second look at, while cleaning up the TransitLayer - and removing all transit data from the graph - we would like to break the OTP into smaller pieces. There should be a transit model, and the global OTP context should have a ref to it and inject it where needed.